src/ParseProxyValues.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Extensions.Primitives;
+
+namespace AspNetCore.ForwardedHttp
+{
+    public class ForwardedValues
+    {
+        public string For { get; set; }
+        public string By { get; set; }
+        public string Host { get; set; }
+        public string Proto { get; set; }
+    }
+
+    //TODO: Need more efficient implementation
+    public class ParseProxyValues
+    {
+        public ParseProxyValues(StringValues values)
```

Comment:
Design: Don't do anything that might fail/throw in a constructor, especially parsing. Make a static Parse method, or better yet a TryParse method.